### PR TITLE
fix: Cosmos proposal by id

### PR DESCRIPTION
**Pull request by mhagel**
## Link to Issue
Closes: #3683

## Description of Changes
- When going to proposals/{proposalId}, if proposal is not loaded into store, load it
- redraw after load

## Test Plan
- CA (click around) tested on local:
  - clear cache before each to ensure API loading:
    - http://localhost:8080/csdk/proposal/2 
    - http://localhost:8080/csdk/proposal/2-title-2
    - http://localhost:8080/osmosis/proposal/2
    - http://localhost:8080/osmosis/proposal/2-set-initial-liquidity-mining-incentives

## Other Considerations

-

### CODE REVIEW

**Review by jnaviask — CHANGES_REQUESTED** [review]
- Loading a proposal directly then navigating to proposals page shows it duplicated in cache.
- Loading a proposal directly then navigating to proposals page does not load remaining old proposals.
2023-05-03-174818_1259x561_scrot

**Comment by mhagel** [comment]
@jnaviask Now replacing existing proposals in the store. This may also solve #3682

File: packages/commonwealth/client/scripts/stores/ProposalStore.ts [hunk 1]
```diff
@@ -5,6 +5,11 @@ class ProposalStore<ProposalT extends IIdentifiable> extends Store<ProposalT> {
   private _storeId: { [hash: string]: ProposalT } = {};
 
   public add(proposal: ProposalT) {
+    const existingProposal = this.getByIdentifier(proposal.identifier);
+    if (existingProposal) {
+      this.remove(existingProposal);
+    }
+
```
Comment (mhagel) [inline packages/commonwealth/client/scripts/stores/ProposalStore.ts:12]: @jnaviask Can you think of any possible side-effects here, maybe in other communities?

Comment (jnaviask) [reply packages/commonwealth/client/scripts/stores/ProposalStore.ts:12]: Nope, this should be fine.

**Comment by jnaviask** [comment]
To recap state: changes here should be merged after #3705 (& rebased against as needed), which solves #3683 but not #3682. Can go out after to solve #3682.

File: packages/commonwealth/client/scripts/views/components/cosmos_proposal_selector/cosmos_proposal_selector.tsx [hunk 2]
```diff
@@ -27,25 +26,21 @@ export const ProposalSelector = ({
...
   const { completedCosmosProposals } = useGetCompletedCosmosProposals({
     app,
-    setIsLoading: setCompletedLoading,
-    isLoading: loadingCompleted,
+    setIsLoading: setCompletedProposalsLoading,
+    isLoading: loadingCompletedProposals,
   });
 
-  useEffect(() => {
-    setLoading(!(activeCosmosProposals.length > 0));
```
Comment (mhagel) [inline packages/commonwealth/client/scripts/views/components/cosmos_proposal_selector/cosmos_proposal_selector.tsx:46]: This was causing infinite bobber if no active proposals. I think my changes here are still what we are looking for.

Comment (jnaviask) [reply packages/commonwealth/client/scripts/views/components/cosmos_proposal_selector/cosmos_proposal_selector.tsx:46]: Confirmed fixed on chihuahua.

**Review by jnaviask — APPROVED** [review]
This is ready to go whenever.

**Pull request merged**